core/src/avm2/globals/xml.rs

```diff
@@ -88,6 +88,50 @@ pub fn to_xml_string<'gc>(
     Ok(Value::String(node.xml_to_xml_string(activation)?))
 }
 
+fn name_to_multiname<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    name: &Value<'gc>,
+) -> Result<Multiname<'gc>, Error<'gc>> {
+    Ok(match name {
+        Value::String(s) => Multiname::new(activation.avm2().public_namespace, *s),
+        Value::Object(o) => {
+            if let Some(qname) = o.as_qname_object() {
+                qname.name().clone()
+            } else {
+                Multiname::new(
+                    activation.avm2().public_namespace,
+                    name.coerce_to_string(activation)?,
+                )
+            }
+        }
+        _ => Multiname::new(
+            activation.avm2().public_namespace,
+            name.coerce_to_string(activation)?,
+        ),
+    })
+}
+
+pub fn child<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let xml = this.unwrap().as_xml_object().unwrap();
+    let multiname = name_to_multiname(activation, &args[0])?;
+    // FIXME: Support numerical indexes.
```

Comment:
I think this should also mention calling `XML.child` with any ("*") as the parameter

Reply:
This is being addressed in a separate PR